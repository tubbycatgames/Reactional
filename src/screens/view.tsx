import React from 'react'
import { FunctionComponent } from 'react'
import { StyleSheet } from 'react-native'

import { SchemeBackground } from '../components/schemed'

export const ScreenView: FunctionComponent = ({ children }) => (
  <SchemeBackground style={styles.container}>{children}</SchemeBackground>
)

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
})
