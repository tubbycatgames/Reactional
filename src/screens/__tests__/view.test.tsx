import React from 'react'
import renderer from 'react-test-renderer'
import { Text } from 'react-native'

import { ScreenView } from '../view'

describe('Screen View', () => {
  it('should wrap children in a screen container', () => {
    const render = renderer.create(
      <ScreenView>
        <Text>Howdy Ho!</Text>
      </ScreenView>
    )
    expect(render.toJSON()).toMatchSnapshot()
  })
})
