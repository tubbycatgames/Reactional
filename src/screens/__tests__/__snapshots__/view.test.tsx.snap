// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen View should wrap children in a screen container 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#B3EFFF",
      },
      Object {
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text>
    Howdy Ho!
  </Text>
</View>
`;
